src/lib/eina/eina_inline_lock_win32.x

```diff
@@ -0,0 +1,767 @@
+        if (ok  == ERROR_SUCCESS) return EINA_LOCK_SUCCEED;
+        mutex->mode = _Eina_RWLock_Mode_Shared;
+     }
+   if (ok == ERROR_ACCESS_DENIED) return EINA_LOCK_FAIL;
+
+   EINA_LOCK_ABORT_DEBUG((int)ok, rwlock_unlock, mutex);
+   return EINA_LOCK_FAIL;
+}
+
+static inline Eina_Bool
+_eina_tls_cb_new(Eina_TLS *key, Eina_TLS_Delete_Cb delete_cb)
+{
+   key = TlsAlloc();
+   DWORD ok = GetLastError();
+   if (key != TLS_OUT_OF_INDEXES || ok != ERROR_SUCCESS) return EINA_TRUE;
+   else return EINA_FALSE;
+}
+
+static inline void
+_eina_tls_free(Eina_TLS key)
+{
+   TlsFree(key);
+}
+
+static inline void *
+_eina_tls_get(Eina_TLS key)
+{
+   return TlsGetValue(key);
+}
+
+static inline Eina_Bool
+_eina_tls_set(Eina_TLS key, const void *data)
+{
+   int ok;
+   DWORD err;
+   ok = TlsSetValue(key, (void *) data);
+   if (ok != 0 && err == ERROR_SUCCESS) return EINA_TRUE;
+   else return EINA_TRUE;
+}
+
+#ifdef EINA_HAVE_WIN32_BARRIER
+struct _Eina_Barrier
+{
+   LPSYNCHRONIZATION_BARRIER barrier;
+};
+#else
+struct _Eina_Barrier
+{
+   int needed;           /**< The number of waiting threads that will cause the barrier to signal and reset. */
+   int called;           /**< The number of threads that are waiting on this barrier. */
+   Eina_Lock cond_lock;  /**< The lock for the barrier */
+   Eina_Condition cond;  /**< The condition variable for the barrier */
+};
+#endif
+
+static inline Eina_Bool
+_eina_barrier_new(Eina_Barrier *barrier, int needed)
+{
+#ifdef EINA_HAVE_WIN32_BARRIER
+   InitializeSynchronizationBarrier(&(barrier->barrier), (LONG) needed, 0);
+   DWORD ok = GetLastError();
+   if (ok == ERROR_SUCCESS) return EINA_TRUE;
+   else if ((ok == ERROR_NOT_ENOUGH_MEMORY) || (ok == ERROR_ACCESS_DENIED)
+            || (ok == ERROR_OUTOFMEMORY))
+    {
+       return EINA_FALSE;
+    }
+   else EINA_LOCK_ABORT_DEBUG(ok, barrier_init, barrier);
+   return EINA_FALSE;
+#else
+   barrier->needed = needed;
+   barrier->called = 0;
+   if (eina_lock_new(&(barrier->cond_lock)))
+     {
+        if (eina_condition_new(&(barrier->cond), &(barrier->cond_lock)))
+          return EINA_TRUE;
+     }
+   return EINA_FALSE;
+#endif
+}
+
+static inline  void
+_eina_barrier_free(Eina_Barrier *barrier)
+{
+#ifdef EINA_HAVE_WIN32_BARRIER
+   DeleteSynchronizationBarrier(&(barrier->barrier));
+   DWORD ok = GetLastError();
+   if (ok != ERROR_SUCCESS) EINA_LOCK_ABORT_DEBUG(ok, barrier_destroy, barrier);
+#else
+   eina_condition_free(&(barrier->cond));
+   eina_lock_free(&(barrier->cond_lock));
+   barrier->needed = 0;
+   barrier->called = 0;
+#endif
+}
+
+static inline Eina_Bool
+_eina_barrier_wait(Eina_Barrier *barrier)
+{
+#ifdef EINA_HAVE_WIN32_BARRIER
+   EnterSyncronizationBarrier(&(barrier->barrier)
+                              , SYNCHRONIZATION_BARRIER_FLAGS_BLOCK_ONLY);
+   DWORD ok = GetLastError();
+   if (ok == ERROR_SUCCESS) return EINA_TRUE;
+   else EINA_LOCK_ABORT_DEBUG((int)ok, barrier_wait, barrier);
+   return EINA_TRUE;
```

Comment:
Is it true for real?

Reply:
It is....You can have a look at both originals `eina_inline_lock_posix.x` and `eina_inline_lock_barrier.x`.

Reply:
Indeed, all paths return true.